### PR TITLE
fix(rolldown): side effect detector mark `+1` `-1` as number literal

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

File: crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs [hunk 1]
```diff
@@ -227,6 +227,12 @@ pub fn is_primitive_literal(scope: &AstScopes, expr: &Expression) -> bool {
     | Expression::NumericLiteral(_)
     | Expression::StringLiteral(_)
     | Expression::BigIntLiteral(_) => true,
+    // oxc number literal did not included `+1` and `-1`
```
Comment (Boshen) [inline crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs]: It's not oxc ... The spec grammar is always an UnaryExpression.

File: crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs [hunk 2]
```diff
@@ -227,6 +227,12 @@ pub fn is_primitive_literal(scope: &AstScopes, expr: &Expression) -> bool {
...
+    Expression::UnaryExpression(e)
+      if matches!(e.operator, |UnaryOperator::UnaryNegation| UnaryOperator::UnaryPlus) =>
```
Comment (Boshen) [inline crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs]: Don't you need to check for number litral on `e.expression`?

File: crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs (hunk 1 above)
Comment (Boshen) [inline crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs:230]: Include `+1` / `-1`.

Comment (Boshen) [inline crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs]:
```suggestion
    // Include `+1` / `-1`.
```

File: crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs (hunk 2 above)
Comment (IWANABETHATGUY) [reply crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs]: you are right, I forgot that.

**Pull request merged**